Tools/environment_install/installROS-ubuntu.sh

```diff
@@ -0,0 +1,230 @@
+{
+    tput setaf 1
+    echo "Your package list did not include a recommended base package"
+    tput sgr0 
+    echo "Please include one of the following:"
+    echo "   ros-base"
+    echo "   desktop"
+    echo "   desktop-full"
+    echo ""
+    echo "ROS not installed"
+}
+
+# Iterate through command line inputs
+packages=()
+while [ "$1" != "" ]; do
+    case $1 in
+        -p | --package )        shift
+                                packages+=("$1")
+                                ;;
+        -h | --help )           usage
+                                exit
+                                ;;
+        * )                     usage
+                                exit 1
+    esac
+    shift
+done
+
+# Checking Ubuntu release to adapt software version to install
+RELEASE_CODENAME=$(lsb_release -c -s)
+PYTHON_V="python3"  # starting from ubuntu 20.04, python isn't symlink to default python interpreter
+PIP=pip3
+
+# echo $RELEASE_CODENAME
+
+if [ ${RELEASE_CODENAME} == 'bionic' ] ; then
+    #Ubuntu 18.04 - Melodic
+    ROS_VERSION="melodic"
+    PYTHON_V="python2"
+    PIP=pip3
+    heading "${green}Detected Ubuntu 18.04, installing ROS Melodic${reset}"
+elif [ ${RELEASE_CODENAME} == 'buster' ]; then
+    #RPi Buster - Melodic
+    ROS_VERSION="melodic"
+    PYTHON_V="python2"
+    PIP=pip3
+    heading "${green}Detected RPi Buster, installing ROS Melodic${reset}"
+elif [ ${RELEASE_CODENAME} == 'focal' ]; then
+    #Ubuntu 20.04 - Noetic
+    ROS_VERSION="noetic"
+    PYTHON_V="python3"
+    PIP=pip3
+    heading "${green}Detected Ubuntu 20.04, installing ROS Noetic${reset}"
+elif [ ${RELEASE_CODENAME} == 'jammy' ]; then
+    #Ubuntu 22.04 - unsupported only ROS2
+    heading "${red}Currently only ROS1 is supported. This Ubuntu release can only be used with ROS2.${reset}"
+    exit 1
+else
+    # We assume an unsupported OS is being used.
+    heading "${red}Unsupported OS detected. Please refer to the ROS webpage to find how to install ROS1 on your system if at all possible.${reset}"
+    exit 1
+fi
+
+# Check to see if other packages were specified
+# If not, set the default base package
+if [ ${#packages[@]}  -eq 0 ] ; then
+ packages+="desktop-full"
+fi
+echo "Packages to install: "${packages[@]}
+# Check to see if we have a ROS base kinda thingie
+hasBasePackage=false
+for package in "${packages[@]}"; do
+  if [[ $package == "ros-base" ]]; then
+     packages=" ros-${ROS_VERSION}-ros-base"
+     hasBasePackage=true
+     break
+  elif [[ $package == "desktop" ]]; then
+     packages=" ros-${ROS_VERSION}-desktop"    
+     hasBasePackage=true
+     break
+  elif [[ $package == "desktop-full" ]]; then
+     packages=" ros-${ROS_VERSION}-desktop-full"    
+     hasBasePackage=true
+     break
+  fi
+done
+if [ $hasBasePackage == false ] ; then
+   shouldInstallPackages
+   exit 1
+fi
+
+heading "${green}Adding Repositories and source lists${reset}"
+#Lets start instaling stuff
+sudo apt-add-repository universe
```

Comment:
need software-properties-common

Reply:
Im not convinced its needed since I have never before manually installed this on any Ubuntu distributiion I have used in the past, but I added it.